pkg/client/cloudevents/source_client.go

```diff
@@ -70,6 +77,16 @@ func NewSourceClient(sourceOptions *ceoptions.CloudEventsSourceOptions, resource
 		})
 	}()
 
+	go func() {
```

Comment:
why in a goroutine? what happens if resync message fails? should we retry?
you might consider select with context so can quite gracefully.

Reply:
Handling the status resync is an asynchronous operation. When the dispatcher detects a new consumer owned by the current Maestro instance, it signals the CloudEvent client to initiate a status resynchronization for the consumer.
This has been updated to use select with context.